Sources/StreamChat/Utils/Logger/Logger.swift

```diff
@@ -178,20 +178,26 @@ public enum LogConfig {
     /// Underlying logger instance to control singleton.
     private static var _logger: Logger?
 
+    private static var queue = DispatchQueue(label: "io.getstream.logconfig")
```

Comment:
You forgot to make this one concurrent

Reply:
This one is the LogConfig,it is not worth making it concurrent, this is called only once when the logger is creatted

Reply:
Besides that we don't use async here

Reply:
are you sure? We have this code:
```swift
queue.async(flags: .barrier) {
       _logger = newValue
}
```

Reply:
Ups! This was a mistake. I've changed this queue to concurrent as well, since we can benefit from concurrent reads here.